frontend/app.py

```diff
@@ -0,0 +1,256 @@
+import streamlit as st
+import http.client
+import pandas as pd
+import plotly.express as px
+from datetime import datetime, timezone, timedelta
+import sys
+import os
+import logging
+import numpy as np
+import xarray as xr
+from dotenv import load_dotenv
+import base64
+import json
+import requests
+from urllib.parse import urlencode
+from PIL import Image
+
+# Load environment variables
+load_dotenv()
+
+# Set up logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Add the parent directory to the Python path
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from quartz_solar_forecast.pydantic_models import PVSite
+from quartz_solar_forecast.forecasts import forecast_v1_tilt_orientation, TryolabsSolarPowerPredictor
+from quartz_solar_forecast.data import get_nwp, process_pv_data
+from quartz_solar_forecast.inverters.enphase import process_enphase_data
+
+# Get the directory of the current script
+script_dir = os.path.dirname(os.path.abspath(__file__))
+
+# Construct the path to logo.png
+logo_path = os.path.join(script_dir, "logo.png")
+im = Image.open(logo_path)
+
+st.set_page_config(page_title="Open Source Quartz Solar Forecast | Open Climate Fix", layout="wide", page_icon=im)
+
+st.title("☀️ Open Source Quartz Solar Forecast")
+
+def get_enphase_auth_url():
```

Comment:
I may be missing something so let me know if I am, is there a reason why we are duplicating a bunch of these functions here and not importing them from the `enphase.py` file?

Reply:
> I may be missing something so let me know if I am, is there a reason why we are duplicating a bunch of these functions here and not importing them from the `enphase.py` file?

This is because some of these functions in enphase.py and even the forecast.py except inputs in the CLI, so reusing them doesn't store the custom inputs, here for example the auth URL, in the browser. So to remove the CLI part entirely to make it browser based I modified those functions a bit

Reply:
> > I may be missing something so let me know if I am, is there a reason why we are duplicating a bunch of these functions here and not importing them from the `enphase.py` file?
> 
> This is because some of the functions in enphase.py and even in forecast.py accept inputs in the CLI, so reusing them doesn't store the custom inputs, here for example the auth URL, in the browser. So to remove the CLI part entirely to make it browser based I modified those functions a bit and rewrote them in the new streamlit file